### PR TITLE
FORMS-3234 Added support to override same values for enum in Dropdown, Check Box Group, and Radio Button component

### PR DESCRIPTION
**Pull request by amanthakur03**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes and the overall coverage did not decrease.
- [ ] All unit tests pass on CircleCi.
- [ ] I ran all tests locally and they pass.

### CODE REVIEW

**Comment by rismehta** [comment]
Build failing on java 8. please check

File: bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/util/AbstractOptionsFieldImpl.java [hunk 1]
```diff
@@ -51,6 +55,30 @@ public boolean isEnforceEnum() {
...
+    /**
+     * Function to override the enum names on providing same enum values
+     *
+     * @return an empty hashmap
+     *         {@code @if} either of the arrays is null or if they have different lengths
+     *
+     *         {@code @else}
+     * @return map containing enum values and enum names as key-value pairs
+     */
+    public Map<Object, String> removeDuplicates() {
```
Comment (rismehta) [inline bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/util/AbstractOptionsFieldImpl.java:67]: This should be private

File: bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/util/AbstractOptionsFieldImpl.java [hunk 2]
```diff
@@ -60,14 +88,18 @@ public Object[] getEnums() {
...
+            String[] enumValue = map.keySet().toArray(new String[0]);
+            return ComponentUtils.coerce(type, enumValue);
         }
     }
 
     @Override
     public String[] getEnumNames() {
         if (enumNames != null) {
-            return Arrays.stream(enumNames)
+            Map<Object, String> map = removeDuplicates();
```
Comment (rismehta) [inline bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/util/AbstractOptionsFieldImpl.java:100]: Please add test cases for this use-case in junit. Also add a cypress test for runtime.

Comment (amanthakur03) [reply bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/util/AbstractOptionsFieldImpl.java:100]: Sure. Thanks

**Pull request closed without merging**